### PR TITLE
ABOH-11: Create CRUD & tests for Habit model

### PR DESCRIPTION
**Pull request by valeryia-rusakova**

### CODE REVIEW

File: project/controllers/habit.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from project.dal.post import PostDAL
+
+
+class HabitController:
+    dal = HabitDAL()
+
+    def get_habit_queryset(self, request, request_type):
+        if request_type == 'list':
+            return self.dal.get_habits_list()
+        habit_pk = request.parser_context['kwargs']['pk']
```
Comment (veranika-zakharchenia) [inline project/controllers/habit.py:12]: what happens if there is no pk provided?

File: project/controllers/habit.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        data = {
+            'name': request.data['name'],
+            'description': request.data['description'],
+            'type': request.data['type'],
+            'image': image_instance,
+        }
+        return self.dal.insert_habit(data)
+
+    def delete_habit(self, request):
+        delete_pk = request.parser_context['kwargs']['pk']
```
Comment (veranika-zakharchenia) [inline project/controllers/habit.py:27]: same here

File: project/serializers.py [hunk 3]
```diff
@@ -1,24 +1,38 @@
...
+        model = Post
+        fields = "__all__"
+
+
+
+
+
+
+
+
```
Comment (veranika-zakharchenia) [inline project/serializers.py:38]: ?

File: project/controllers/habit.py (hunk 1 above)
Comment (valeryia-rusakova) [reply project/controllers/habit.py:11]: it's impossible, we check list request higher, so if pk is not provided we'll get list of records instead single record

File: project/controllers/habit.py (hunk 2 above)
Comment (valeryia-rusakova) [reply project/controllers/habit.py:26]: this method can't be called without pk in request

**Review by veranika-zakharchenia — APPROVED** [review]

**Pull request merged**